aio/aio-proxy/aio_proxy/routes.py

```diff
@@ -33,29 +37,19 @@
 
 
 @routes.get("/search")
-async def search_endpoint(request):
-    try:
-        terms, page, per_page, filters = extract_parameters(request)
-        total_results, entreprises = search_es(
-            Siren, terms, page * per_page, per_page, **filters
-        )
-        unite_legale = hide_fields(entreprises)
-        res = {
-            "unite_legale": unite_legale,
-            "total_results": int(total_results),
-            "page": page + 1,
-            "per_page": per_page,
-        }
-        res["total_pages"] = int(res["total_results"] / res["per_page"]) + 1
-        return web.json_response(text=json.dumps([res], default=str))
-    except (elasticsearch.exceptions.RequestError, ValueError, TypeError) as error:
-        raise web.HTTPBadRequest(
-            text=serialize_error_text(str(error)), content_type="application/json"
-        )
-    except BaseException as error:
-        raise web.HTTPInternalServerError(
-            text=serialize_error_text(str(error)), content_type="application/json"
-        )
+async def search_text_endpoint(request):
+    response = api_response(
+        request, extract_function=extract_text_parameters, search_function=search_text
+    )
+    return web.json_response(text=json.dumps([response], default=str))
+
+
+@routes.get("/near_point")
+async def near_point_endpoint(request):
+    response = api_response(
+        request, extract_function=extract_geo_parameters, search_function=search_geo
+    )
+    return web.json_response(text=json.dumps([response], default=str))
```

Comment:
Great attempt at factorizing error management but smthg bothers me : `web.json_response(text=json.dumps([response], default=str))` can definitely be bugged, and trigger a 500 that would be text formatted